### PR TITLE
Merge JCore inside the generator

### PR DESCRIPTION
**Pull request by MathieuAA**
### What it does:
  - Adds the commits from JCore into this repo
  - Moves the relevant files to a folder named `jdl` inside the repo
    - does the same for tests
  - Renames the files to match the naming convention here
  - Updates the imported tests
  - Updates the test:unit command in the `package.json` file to use the mocha config file
    - results in a cleaner package.json "scripts" section

### What it doesn't do:
  - Include breaking changes, or any change to existing files
  - Refactor the already present files to use the newer files (another PR is coming, later)
  - Remove JCore as a dependency just **yet** (another PR is coming, later)

### Next steps:
  - Archive the JCore repository
    - ping PR submitters to head over here for their work
  - Do what hasn't been done earlier: refactoring :)

### CODE REVIEW

**Comment by deepu105** [comment]
@MathieuAA @jdubois @pascalgrimaud should we merge this or wait for 7.x freeze?

File: .vscode/launch.json [hunk 1]
```diff
@@ -1,292 +0,0 @@
-{
-    // Use IntelliSense to learn about possible attributes.
-    // Hover to view descriptions of existing attributes.
```
Comment (deepu105) [inline .vscode/launch.json:3]: we shouldn't delete this, I hope it was deleted by mistake

**Comment by pascalgrimaud** [comment]
@deepu105 : absolutely not now ! :-D Plz, don't afraid me like this lol.
We should wait for v7 as it will change the "release mode" and when it will be merged, I'll work with @MathieuAA for changing this process

**Comment by deepu105** [comment]
@pascalgrimaud no worries was just checking :wink: also I totally overlooked the draft status

**Comment by MathieuAA** [comment]
"Press the big red button"
"Press the big red button"

File: .vscode/launch.json (hunk 1 above)
Comment (MathieuAA) [reply .vscode/launch.json:3]: I'll get it back, thanks for pointing that out!

**Comment by MathieuAA** [comment]
@deepu105 I don't know about the naming. I chose JDL because a lot of what's inside is JDL-related. What's not JDL-related is JHipster-related (like field types for instance). The minor part can be moved elsewhere though

File: .vscode/launch.json (hunk 1 above)
Comment (MathieuAA) [reply .vscode/launch.json]: Done

**Comment by MathieuAA** [comment]
Alright! as v7 is the next step, this PR is no longer a draft.

**Comment by pascalgrimaud** [comment]
I'd like to merge this, before going further in v7
Is it ready for beeing merged @MathieuAA ?

**Comment by MathieuAA** [comment]
Unless someone wants to review it first, yes

**Comment by pascalgrimaud** [comment]
As mentionned, no breaking changes here, as it's the merge of 2 projects
Then let's go !

**Comment by MathieuAA** [comment]
I don't know if you can do that @pascalgrimaud but now that the merge is live, we should archive JCore

**Comment by pascalgrimaud** [comment]
@MathieuAA : I'd prefer to keep jhipster-core like this, and archive it once v7 is released. The reason is, in case we find a bug and we need to do new release of jhipster-core and new patch release of generator-jhipster

**Pull request merged**